src/Microsoft.SqlTools.Hosting/Properties/AssemblyInfo.cs

```diff
@@ -49,3 +49,4 @@
 [assembly: InternalsVisibleTo("MicrosoftSqlToolsServiceLayer")]
 [assembly: InternalsVisibleTo("MicrosoftKustoServiceLayer")]
 [assembly: InternalsVisibleTo("MicrosoftSqlToolsCredentials")]
+[assembly: InternalsVisibleTo("SqlToolsResourceProviderService")]
```

Comment:
What is this needed for? Generally we should try to avoid this - if something needs to be accessed by other assemblies just make it public.

Reply:
This is meant to access the functions for CommandOptions, such as LocaleSetter, which throws up an access error if the internals are not visible. 

See:
https://github.com/microsoft/sqltoolsservice/blob/alex/localizationbuild/src/Microsoft.SqlTools.ServiceLayer/Utility/ServiceLayerCommandOptions.cs

and 

https://github.com/microsoft/sqltoolsservice/blob/alex/localizationbuild/src/Microsoft.Kusto.ServiceLayer/Utility/ServiceLayerCommandOptions.cs

For examples

Reply:
Where is LocaleSetter defined? If we could just make that public I'd much prefer that - InternalsVisibleTo isn't really a great pattern to get in the habit of using since it makes determining dependencies much more difficult.

Reply:
It is defined here: https://github.com/microsoft/sqltoolsservice/blob/alex/localizationbuild/src/Microsoft.SqlTools.Hosting/Utility/CommandOptions.cs However, turning the function from internal to public doesn't work as the ResourceCommandOptions has no context for LocaleSetter without the InternalsVisible line.

Reply:
I set LocaleSetter to public and also removed internalsvisible for credentials and kusto, (ran the unit tests on them without issue, but I can restore them if necessary, ServiceLayer still needs it though).